File: src/main/java/com/googongill/aditory/service/dto/category/CopyCategoryResult.java
package com.googongill.aditory.service.dto.category;

import com.googongill.aditory.domain.Category;
import com.googongill.aditory.domain.enums.CategoryState;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class CopyCategoryResult {
    private Long categoryId;
    private String categoryName;
    private CategoryState categoryState;
    private LocalDateTime createdAt;

    public static CopyCategoryResult of(Category newCategory) {
        return CopyCategoryResult.builder()
                .categoryId(newCategory.getId())
                .categoryName(newCategory.getCategoryName())
                .categoryState(newCategory.getCategoryState())
                .createdAt(newCategory.getCreatedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/category/CopyCategoryResult.kt
package com.googongill.aditory.service.dto.category

import com.googongill.aditory.domain.Category
import com.googongill.aditory.domain.enums.CategoryState
import java.time.LocalDateTime

data class CopyCategoryResult(
    val categoryId: Long?,
    val categoryName: String,
    val asCategoryName: String,
    val categoryState: CategoryState,
    val createdAt: LocalDateTime?
) {

    companion object {
        fun of(newCategory: Category): CopyCategoryResult {
            return CopyCategoryResult(
                categoryId = newCategory.id,
                categoryName = newCategory.categoryName,
                asCategoryName = newCategory.asCategoryName,
                categoryState = newCategory.categoryState,
                createdAt = newCategory.createdAt
            )
        }
    }

}